Casper.Network.SDK.Test/DurationTest.cs

```diff
@@ -0,0 +1,119 @@
+                            + 10_000
+                            + 31,
+                DurationUtils.StringToMilliseconds("2days 2h 30m 10s 31ms"));
+
+            Assert.AreEqual(172_800_000 * 365.25d
+                            + 7 * 86_400_000d * 30.44d
+                            + 86_400_000
+                            + 10_000
+                            + 31,
+                DurationUtils.StringToMilliseconds("2years 7 months 1 day 10s 31ms"));
+
+            Assert.AreEqual(7 * 24 * 3600 * 1000,
+                DurationUtils.StringToMilliseconds("1week"));
+            Assert.AreEqual(7 * 24 * 3600 * 1000,
+                DurationUtils.StringToMilliseconds("1weeks"));
+            Assert.AreEqual(3 * 7 * 24 * 3600 * 1000,
+                DurationUtils.StringToMilliseconds("3weeks"));
+
+            Assert.AreEqual(0, DurationUtils.StringToMilliseconds("0years 0days 0s"));
+        }
+
+        [Test]
+        public void CatchWrongDurationTest()
+        {
+            var ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.MillisecondsToString(-1000d));
+            Assert.IsNotNull(ex);
+            Assert.IsTrue(ex.Message.Contains("Negative values not allowed."));
+
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.MillisecondsToString(double.MaxValue));
+            Assert.IsNotNull(ex);
+
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("584542047years"));
+            Assert.IsNotNull(ex);
+
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("99999999999999999999999999ms"));
+            Assert.IsNotNull(ex);
+
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("-1ms"));
+            Assert.IsNotNull(ex);
+
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("ms"));
+            Assert.IsNotNull(ex);
+
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("xms"));
+            Assert.IsNotNull(ex);
+
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("7 eras"));
+            Assert.IsNotNull(ex);
+
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("123"));
+            Assert.IsNotNull(ex);
+            
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("1m 7"));
+            Assert.IsNotNull(ex);
+            
+            ex = Assert.Catch<ArgumentOutOfRangeException>(() =>
+                DurationUtils.StringToMilliseconds("1m 7 1ms"));
+            Assert.IsNotNull(ex);
+        }
+    }
+}
```

Comment:
Is this a conscious decision? If not, can we have a newline at the end to play nice with *nix based systems/editors? (Same for other files in the PR.)

Reply:
I checked and most of the source code files do not have a newline a the end.
I've configured my IDE to ensure a new line is added to the end on save.
I propose to have a separate PR to fix this, since there'll be dozens of files with only this change.

Reply:
Sounds good to me.